crates/cxx-qt-lib/src/gui/qimage.rs

```diff
@@ -332,6 +371,73 @@ impl QImage {
             None
         }
     }
+
+    /// Constructs a QImage from an existing memory buffer.
+    ///
+    /// # Safety
+    /// For details on safety see the [Qt documentation](https://doc.qt.io/qt-6/qimage.html#QImage-7)
+    pub unsafe fn from_raw_parts(
+        data: *const ffi::uchar,
+        width: i32,
+        height: i32,
+        format: QImageFormat,
+        cleanup_function: extern "C" fn(x: *mut ffi::c_void),
+        cleanup_info: *mut ffi::c_void,
+    ) -> Self {
+        ffi::qimage_init_from_raw_parts(
+            data,
+            width,
+            height,
+            format,
+            QImageCleanupFunction(cleanup_function),
+            cleanup_info,
+        )
+    }
+
+    /// Constructs a QImage from an existing mutable memory buffer.
+    ///
+    /// # Safety
+    /// For details on safety see the [Qt documentation](https://doc.qt.io/qt-6/qimage.html#QImage-8)
+    pub unsafe fn from_raw_parts_mut(
+        data: *mut ffi::uchar,
+        width: i32,
+        height: i32,
+        format: QImageFormat,
+        cleanup_function: extern "C" fn(x: *mut ffi::c_void),
+        cleanup_info: *mut ffi::c_void,
+    ) -> Self {
+        ffi::qimage_init_from_raw_parts_mut(
+            data,
+            width,
+            height,
+            format,
+            QImageCleanupFunction(cleanup_function),
+            cleanup_info,
+        )
+    }
+
+    /// Constructs a QImage from the raw data inside a `Vec<u8>`.
+    ///
+    /// # Safety
+    /// This function is unsafe because it requires that the data matches the given QImageFormat,
+    /// width and height.
+    ///
+    /// It is a convenience function around [Self::from_raw_parts_mut].
+    pub unsafe fn from_raw_bytes(
+        data: Vec<u8>,
+        width: i32,
+        height: i32,
+        format: QImageFormat,
+    ) -> Self {
+        extern "C" fn delete_boxed_vec(boxed_vec: *mut ffi::c_void) {
+            drop(Box::new(boxed_vec as *mut Vec<u8>));
```

Comment:
this looks bogus to me, can you explain it please? isn't `Box::new()` like a `std::make_unique`, meaning it does a new memory allocation? Or is this a misnomer and it actually adopts the ptr that gets passed in, so more like `std::unique_ptr<T>(...)`?

Reply:
Very good catch, thank you!

It needs to be `Box::from_raw`. Should be fixed now.

Reply:
Curious that valgrind didn't spot this, or are only the Qt unit tests not the `cargo test` run under valgrind ? Maybe this should be added to the qt_types_standalone or we should run the `cargo test` (even just for cxx-qt-lib) under valgrind ? :thinking:

Reply:
I just checked, it indeed did not find the issue.
It seems to be that we really don't run `cargo test` under valgrind, but just the tests that we run via C++ (e.g. qml_features_test, etc.)

Reply:
We could try using: https://github.com/jfrimmel/cargo-valgrind
But that would probably be better served in a different PR, as that means we might want to remove/reduce the qt_types_standalone tests as well :thinking:

Reply:
as an outsider: can you use sanitizers instead? they are much faster and also have leak checking built-in - that should be preferred over valgrind nowadays

Reply:
Done as #1177